File: python/appwrite/services/general.py
from ..service import Service


class General(Service):

    def __init__(self, client):
        super(General, self).__init__(client)

    def empty(self):
        """Mock a post request for SDK tests"""

        params = {}
        path = '/mock/tests/general/empty'

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def redirect(self):
        """Mock a post request for SDK tests"""

        params = {}
        path = '/mock/tests/general/redirect'

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def redirected(self):
        """Mock a post request for SDK tests"""

        params = {}
        path = '/mock/tests/general/redirected'

        return self.client.call('get', path, {
            'content-type': 'application/json',
        }, params)

    def upload(self, x, y, z, file):
        """Mock a post request for SDK tests"""

        params = {}
        path = '/mock/tests/general/upload'
        params['x'] = x
        params['y'] = y
        params['z'] = z
        params['file'] = file

        return self.client.call('post', path, {
            'content-type': 'multipart/form-data',
        }, params)

File: node/lib/services/general.js
const Service = require('../service.js');

class General extends Service {

    /**
     * Mock a post request for SDK tests
     *
     * @throws Exception
     * @return {}
     */
    async empty() {
        let path = '/mock/tests/general/empty';
        
        return await this.client.call('get', path, {
                    'content-type': 'application/json',
               },
               {
            });
    }

    /**
     * Mock a post request for SDK tests
     *
     * @throws Exception
     * @return {}
     */
    async redirect() {
        let path = '/mock/tests/general/redirect';
        
        return await this.client.call('get', path, {
                    'content-type': 'application/json',
               },
               {
            });
    }

    /**
     * Mock a post request for SDK tests
     *
     * @throws Exception
     * @return {}
     */
    async redirected() {
        let path = '/mock/tests/general/redirected';
        
        return await this.client.call('get', path, {
                    'content-type': 'application/json',
               },
               {
            });
    }

    /**
     * Mock a post request for SDK tests
     *
     * @param string x
     * @param number y
     * @param array z
     * @param File file
     * @throws Exception
     * @return {}
     */
    async upload(x, y, z, file) {
        let path = '/mock/tests/general/upload';
        
        return await this.client.call('post', path, {
                    'content-type': 'multipart/form-data',
               },
               {
                'x': x,
                'y': y,
                'z': z,
                'file': file
            });
    }
}

module.exports = General;